mava/wrappers/jumanji.py

```diff
@@ -34,26 +35,58 @@
 
 
 class MultiAgentWrapper(Wrapper):
-    def __init__(self, env: Environment):
+    def __init__(self, env: Environment, add_global_state: bool = False):
         super().__init__(env)
         self._num_agents = self._env.num_agents
         self.time_limit = self._env.time_limit
+        self.add_global_state = add_global_state
 
+    @abstractmethod
     def modify_timestep(self, timestep: TimeStep) -> TimeStep[Observation]:
         """Modify the timestep for `step` and `reset`."""
         pass
 
+    def get_global_state(self, obs: Observation) -> chex.Array:
+        """The default way to create a global state for an environment if it has no
+        available global state - concatenate all observations
+        """
+        global_state = jnp.concatenate(obs.agents_view, axis=0)
+        global_state = jnp.tile(global_state, (self._env.num_agents, 1))
+        return global_state
+
     def reset(self, key: chex.PRNGKey) -> Tuple[State, TimeStep]:
         """Reset the environment."""
         state, timestep = self._env.reset(key)
-        return state, self.modify_timestep(timestep)
+        timestep = self.modify_timestep(timestep)
+        if self.add_global_state:
+            global_state = self.get_global_state(timestep.observation)
+            observation = ObservationGlobalState(
+                global_state=global_state,
+                agents_view=timestep.observation.agents_view,
+                action_mask=timestep.observation.action_mask,
+                step_count=timestep.observation.step_count,
+            )
```

Comment:
The problem with moving into `get_global_obs` is that it will have to be repeated every time a wrapper overwrites the method. So I think the current way has less code duplication.